tests/FSharpPlus.Tests/General.fs

```diff
@@ -2522,3 +2522,16 @@ module testCompileOldCode =
         let rec f a b n = if n == 0G then a else f (b * a) b (n - 1G)
         if (n < 0G) then failwith "Negative exponent" else f 1G x n
     let inline ( **^^ ) (x:'Fractional) (n:'Integral) = if n >= 0G then x**^n else recip (x**^(negate n))
+
+module choosei =
+    [<Test>]
+    let choosei () =
+        let someIfIndexEven i x =
+            if i % 2 = 0 then Some x
+            else None
+        (choosei someIfIndexEven [1;2;3;4;5]) |> areEqual [1;3;5]
```

Comment:
These tests are ok, but I would be more interested in tests that checks the right overloads are called, see for reference tests of other Indexables where we ask IsInstanceOf.

Reply:
Added tests for instance types, you mean something like this?